cmd/autotestgridgenerator/main.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/experiment/autobumper/bumper"
+	"k8s.io/test-infra/prow/config/secret"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/labels"
+)
+
+const (
+	githubOrg    = "kubernetes"
+	githubRepo   = "test-infra"
+	githubLogin  = "openshift-bot"
+	githubTeam   = "openshift/openshift-team-developer-productivity-test-platform"
+	matchTitle   = "Automate testgrid generator"
+	remoteBranch = "auto-testgrid-generator"
+)
+
+type options struct {
+	selfApprove bool
+
+	githubLogin       string
+	gitName           string
+	gitEmail          string
+	targetDir         string
+	assign            string
+	testgridConfigDir string
+	releaseConfigDir  string
+	prowJobsDir       string
+	allowList         string
+
+	flagutil.GitHubOptions
+}
+
+func parseOptions() options {
+	var o options
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.githubLogin, "github-login", githubLogin, "The GitHub username to use.")
+	fs.StringVar(&o.gitName, "git-name", "", "The name to use on the git commit. Requires --git-email. If not specified, uses the system default.")
+	fs.StringVar(&o.gitEmail, "git-email", "", "The email to use on the git commit. Requires --git-name. If not specified, uses the system default.")
+	fs.StringVar(&o.targetDir, "target-dir", "", "The directory containing the target repo.")
+	fs.StringVar(&o.assign, "assign", githubTeam, "The github username or group name to assign the created pull request to.")
+	fs.StringVar(&o.testgridConfigDir, "testgrid-config", "../../../../kubernetes/test-infra/config/testgrids/openshift", "The directory where the testgrid output will be stored")
+	fs.StringVar(&o.releaseConfigDir, "release-config", "../../../release/core-services/release-controller/_releases", "The directory of release config files")
+	fs.StringVar(&o.prowJobsDir, "prow-jobs-dir", "../../../release/ci-operator/jobs", "The directory where prow-job configs are stored")
+	fs.StringVar(&o.allowList, "allow-list", "../../../release/core-services/testgrid-config-generator/_allow-list.yaml", "File containing release-type information to override the defaults")
+
+	fs.BoolVar(&o.selfApprove, "self-approve", false, "Self-approve the PR by adding the `approved` and `lgtm` labels. Requires write permissions on the repo.")
+	o.AddFlags(fs)
+	o.AllowAnonymous = true
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Errorf("cannot parse args: '%s'", os.Args[1:])
+	}
+	return o
+}
+
+func validateOptions(o options) error {
+	if o.githubLogin == "" {
+		return fmt.Errorf("--github-login cannot be empty string")
+	}
+	if (o.gitEmail == "") != (o.gitName == "") {
```

Comment:
Reading this, I realize that we reach a point that we have a lot of `auto*` tools and this code is duplicated everywhere. Maybe it's better to generalize those options and its validation and import it in this tool and update the rest of the tools in a follow-up PR.

Reply:
That was my point at the very start about having these as separate tools that call `os.Command` but you and @alvaroaleman disagreed :trollface:

Reply:
This exists now in `pkg/github/prcreation`